apps/custom-config/wakeup-effect.js

```diff
@@ -254,11 +255,13 @@ class WakeupEffect extends BaseConfig {
           if (typeof queryObj.wakeupSoundEffects !== 'object') {
             return
           }
-          clearDir(ActivationConfig.customPath).then(() => {
-            downloadWav(queryObj.wakeupSoundEffects, ActivationConfig.customPath).then((fileList) => {
-              this.notifyActivation(fileList)
-            }).catch((err) => {
-              logger.warn(`download custom wakeup sound error: ${err}`)
+          mkdirp(ActivationConfig.customPath, () => {
+            clearDir(ActivationConfig.customPath).then(() => {
+              downloadWav(queryObj.wakeupSoundEffects, ActivationConfig.customPath).then((fileList) => {
+                this.notifyActivation(fileList)
+              }).catch((err) => {
+                logger.warn(`download custom wakeup sound error: ${err}`)
+              })
             })
```

Comment:
There is no need to nest these promises.

Reply:
Are file operations synchronous?